core/src/main/java/org/mule/runtime/core/privileged/exception/OnRuntimeProcessingStrategy.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.privileged.exception;
+
+import org.mule.runtime.api.component.location.ConfigurationComponentLocator;
+import org.mule.runtime.api.component.location.Location;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.context.notification.FlowStackElement;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+import org.mule.runtime.core.api.processor.Sink;
+import org.mule.runtime.core.api.processor.strategy.ProcessingStrategy;
+import org.mule.runtime.core.privileged.processor.MessageProcessors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+
+/**
+ * A {@link ProcessingStrategy} used for {@link org.mule.runtime.core.internal.exception.GlobalErrorHandler}. The resolution of
+ * the processing strategy is done in runtime depending on the flow location where the error that is being handled was thrown.
+ */
+public class OnRuntimeProcessingStrategy implements ProcessingStrategy {
+
+  private static final Logger logger = LoggerFactory.getLogger(OnRuntimeProcessingStrategy.class);
+
+  private final ConfigurationComponentLocator locator;
+
+  public OnRuntimeProcessingStrategy(ConfigurationComponentLocator locator) {
+    this.locator = locator;
+  }
+
+  @Override
+  public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor pipeline) {
+    return null;
```

Comment:
better throw an UnsupportedOperationException. Add a comment explaining the reason for this